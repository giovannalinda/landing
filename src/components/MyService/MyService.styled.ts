import styled from 'styled-components'
import { theme } from '~/styles'

export const Container = styled.section`
  padding-top: 18.8rem;
  padding-bottom: 8rem;
`

export const ServicesList = styled.ul`
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-gap: ${theme.space[5]};
  margin-top: 7rem;
`
