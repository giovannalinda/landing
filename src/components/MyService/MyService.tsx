import { FiCode, FiLayers, FiColumns, FiBox } from 'react-icons/fi'
import { InView } from 'react-intersection-observer'

import { Title } from '~/components'
import { ServiceCard } from './ServiceCard'

import { useTranslation } from '~/lib/useTranslation'

import * as S from './MyService.styled'

export function MyService() {
  const { t } = useTranslation()

  return (
    <InView triggerOnce>
      {({ inView, ref }) => (
        <S.Container id='services' ref={ref}>
          <Title>{t('My Service')}</Title>

          {inView && (
            <S.ServicesList>
              <li>
                <ServiceCard title='UI/UX Design' delay={0}>
                  <FiLayers size={40} strokeWidth={1} />
                </ServiceCard>
              </li>
              <li>
                <ServiceCard title='Product Design' delay={0.3}>
                  <FiBox size={40} strokeWidth={1} />
                </ServiceCard>
              </li>
              <li>
                <ServiceCard title='Branding Design' delay={0.6}>
                  <FiColumns size={35} strokeWidth={1} />
                </ServiceCard>
              </li>
              <li>
                <ServiceCard title='Front End Development' delay={0.9}>
                  <FiCode size={40} strokeWidth={1} />
                </ServiceCard>
              </li>
            </S.ServicesList>
          )}
        </S.Container>
      )}
    </InView>
  )
}
