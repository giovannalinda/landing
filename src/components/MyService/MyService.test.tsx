import { render } from '@testing-library/react'

import { MyService } from './MyService'

describe('MyService', () => {
  it('should match snapshot', () => {
    const { container } = render(<MyService />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
