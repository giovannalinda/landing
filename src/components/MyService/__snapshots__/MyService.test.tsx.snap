// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyService should match snapshot 1`] = `
<section
  class="sc-dmRaPn iUJwkw"
>
  <h1
    class="sc-breuTD cNMyLJ"
  >
    My Service
  </h1>
  <hr
    class="sc-ksZaOG ixoBSr"
  />
  <ul
    class="sc-kgflAQ jQnlXa"
  >
    <li>
      <button
        aria-label="UI/UX Design"
        class="sc-idiyUo cMnzpJ"
      >
        <svg
          fill="none"
          height="40"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1"
          viewBox="0 0 24 24"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polygon
            points="12 2 2 7 12 12 22 7 12 2"
          />
          <polyline
            points="2 17 12 22 22 17"
          />
          <polyline
            points="2 12 12 17 22 12"
          />
        </svg>
        <div
          class="sc-dIouRR buHGXV"
        >
          UI/UX Design
          <hr
            class="sc-hHLeRK fOcXOD"
          />
          <div
            class="sc-gsnTZi bljDYF"
          >
            <span>
              Discuss now
            </span>
            <svg
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M23.0677 11.9929L18.818 7.75739L17.4061 9.17398L19.2415 11.0032L0.932469 11.0012L0.932251 13.0012L19.2369 13.0032L17.4155 14.8308L18.8321 16.2426L23.0677 11.9929Z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
      </button>
    </li>
    <li>
      <button
        aria-label="Product Design"
        class="sc-idiyUo cMnzpJ"
      >
        <svg
          fill="none"
          height="40"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1"
          viewBox="0 0 24 24"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21 16V8a2 2 0 0 0-1-1.73l-7-4a2 2 0 0 0-2 0l-7 4A2 2 0 0 0 3 8v8a2 2 0 0 0 1 1.73l7 4a2 2 0 0 0 2 0l7-4A2 2 0 0 0 21 16z"
          />
          <polyline
            points="3.27 6.96 12 12.01 20.73 6.96"
          />
          <line
            x1="12"
            x2="12"
            y1="22.08"
            y2="12"
          />
        </svg>
        <div
          class="sc-dIouRR buHGXV"
        >
          Product Design
          <hr
            class="sc-hHLeRK fOcXOD"
          />
          <div
            class="sc-gsnTZi bljDYF"
          >
            <span>
              Discuss now
            </span>
            <svg
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M23.0677 11.9929L18.818 7.75739L17.4061 9.17398L19.2415 11.0032L0.932469 11.0012L0.932251 13.0012L19.2369 13.0032L17.4155 14.8308L18.8321 16.2426L23.0677 11.9929Z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
      </button>
    </li>
    <li>
      <button
        aria-label="Branding Design"
        class="sc-idiyUo cMnzpJ"
      >
        <svg
          fill="none"
          height="35"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1"
          viewBox="0 0 24 24"
          width="35"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 3h7a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2h-7m0-18H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h7m0-18v18"
          />
        </svg>
        <div
          class="sc-dIouRR buHGXV"
        >
          Branding Design
          <hr
            class="sc-hHLeRK fOcXOD"
          />
          <div
            class="sc-gsnTZi bljDYF"
          >
            <span>
              Discuss now
            </span>
            <svg
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M23.0677 11.9929L18.818 7.75739L17.4061 9.17398L19.2415 11.0032L0.932469 11.0012L0.932251 13.0012L19.2369 13.0032L17.4155 14.8308L18.8321 16.2426L23.0677 11.9929Z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
      </button>
    </li>
    <li>
      <button
        aria-label="Front End Development"
        class="sc-idiyUo cMnzpJ"
      >
        <svg
          fill="none"
          height="40"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1"
          viewBox="0 0 24 24"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="16 18 22 12 16 6"
          />
          <polyline
            points="8 6 2 12 8 18"
          />
        </svg>
        <div
          class="sc-dIouRR buHGXV"
        >
          Front End Development
          <hr
            class="sc-hHLeRK fOcXOD"
          />
          <div
            class="sc-gsnTZi bljDYF"
          >
            <span>
              Discuss now
            </span>
            <svg
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M23.0677 11.9929L18.818 7.75739L17.4061 9.17398L19.2415 11.0032L0.932469 11.0012L0.932251 13.0012L19.2369 13.0032L17.4155 14.8308L18.8321 16.2426L23.0677 11.9929Z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
      </button>
    </li>
  </ul>
</section>
`;
